okhttp/src/test/java/okhttp3/PlatformRule.kt

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2019 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okhttp3.internal.platform.Platform
+import org.conscrypt.Conscrypt
+import org.hamcrest.CoreMatchers.equalTo
+import org.hamcrest.CoreMatchers.not
+import org.junit.Assume
+import org.junit.Assume.assumeThat
+import org.junit.rules.ExternalResource
+import java.security.Security
+
+/**
+ * Marks a test as Platform aware, before the test runs a consistent Platform will be
+ * established e.g. SecurityManager for Conscrypt installed.
+ *
+ * Also allows a test file to state general platform assumptions, or for individual test.
+ */
+open class PlatformRule @JvmOverloads constructor(
```

Comment:
We may want to extend this to allow testing the failure modes on specific platforms instead.  Instead of failing an assumption, we could verify that it fails with a specific error in Conscrypt.  Then if Conscrypt starts working, we would detect the regression.

Reply:
Cool!